app/questionnaire/questionnaire_schema.py

```diff
@@ -215,6 +215,12 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    @staticmethod
+    def _check_list_exists(rule: Union[dict, tuple], list_name: str) -> bool:
+        return isinstance(rule, dict) and (
+            rule.get("list") == list_name or rule.get("identifier") == list_name
+        )
```

Comment:
Which would this be a tuple?. Wouldn't the existing code error if this was a tuple and not a mapping?

Reply:
as discussed yesterday, you were right, it wasn't picking up nested rule. I have fixed it now.